### PR TITLE
Move KeyValueIterable and KeyValueIterableView from trace to common

### PR DESCRIPTION
**Pull request by xukaren**
A simple PR based on [this comment]( ) from @maxgolov.

Since `KeyValueIterable` is used by metrics, traces, and possibly later on by logs, it would be better if it belonged to the “common” namespace as it is no longer exclusive to being used for “trace”.  

In this PR, `KeyValueIterable` and its derived class `KeyValueIterableView`, are refactored from namespace `trace::*` to `common::*` and their corresponding  `*.h` files are moved from `api/.../trace` to `api/.../common`. 

cc @alolita

### CODE REVIEW

File: api/test/metrics/noop_metrics_test.cc [hunk 1]
```diff
@@ -43,7 +43,7 @@ TEST(NoopMeter, RecordBatch)
   std::unique_ptr<Meter> m{std::unique_ptr<Meter>(new NoopMeter{})};
 
   std::map<std::string, std::string> labels = {{"Key", "Value"}};
-  auto labelkv = opentelemetry::trace::KeyValueIterableView<decltype(labels)>{labels};
+  auto labelkv = opentelemetry::common::KeyValueIterableView<decltype(labels)>{labels};
```
Comment (lalitb) [inline api/test/metrics/noop_metrics_test.cc:46]: Not related to this PR, but seems we need to be consistent with the scope resolution usage - either use `opentelemetry::common::KeyValueIterableView` or `common::KeyValueIterableView`. 

Nothing wrong with this PR thought. This needs to be fixed as separate PR for all such entities.

**Review by lalitb — APPROVED** [review]

File: ext/include/opentelemetry/ext/zpages/threadsafe_span_data.h [hunk 2]
```diff
@@ -175,8 +175,9 @@ class ThreadsafeSpanData final : public opentelemetry::sdk::trace::Recordable
   void AddEvent(
       nostd::string_view name,
       core::SystemTimestamp timestamp = core::SystemTimestamp(std::chrono::system_clock::now()),
-      const trace_api::KeyValueIterable &attributes =
-          trace_api::KeyValueIterableView<std::map<std::string, int>>({})) noexcept override
+      const opentelemetry::common::KeyValueIterable &attributes =
+          opentelemetry::common::KeyValueIterableView<std::map<std::string, int>>(
```
Comment (pyohannes) [inline ext/include/opentelemetry/ext/zpages/threadsafe_span_data.h:179]: With this change, can we get use of the `namespace trace_api = opentelemetry::trace;` at the top of this file?

File: sdk/include/opentelemetry/sdk/trace/recordable.h [hunk 3]
```diff
@@ -83,7 +83,7 @@ class Recordable
    * @param attributes the attributes associated with the link
    */
   virtual void AddLink(opentelemetry::trace::SpanContext span_context,
-                       const trace_api::KeyValueIterable &attributes) noexcept = 0;
+                       const opentelemetry::common::KeyValueIterable &attributes) noexcept = 0;
```
Comment (pyohannes) [inline sdk/include/opentelemetry/sdk/trace/recordable.h:86]: Same here: could we get rid of the `trace_api` definition?

**Review by pyohannes — APPROVED** [review]
Thanks for this work, this looks good.

It makes perfect sense to get rid of dependencies from the `metrics` to the `trace` part.

File: ext/include/opentelemetry/ext/zpages/threadsafe_span_data.h (hunk 2 above)
Comment (xukaren) [reply ext/include/opentelemetry/ext/zpages/threadsafe_span_data.h:179]: Did you mean to get remove the line `namespace trace_api = opentelemetry::trace;`? I think trace_api:: is still being used by`trace_api::CanonicalCode` in line 140. Would you prefer it be removed and use `opentelemetry::trace::CanonicalCode` instead?

**Comment by reyang** [comment]
CLA is broken today.

**Review by maxgolov — APPROVED** [review]

File: sdk/include/opentelemetry/sdk/trace/recordable.h (hunk 3 above)
Comment (ThomsonTan) [reply sdk/include/opentelemetry/sdk/trace/recordable.h:86]: Create generic short pattern for `opentelemetry::common` as `common_api` or even as `api`?

**Review by ThomsonTan — APPROVED** [review]

File: sdk/include/opentelemetry/sdk/trace/recordable.h (hunk 3 above)
Comment (xukaren) [reply sdk/include/opentelemetry/sdk/trace/recordable.h:86]: Sure thing - should I leave `trace_api` in as well then?

**Comment by maxgolov** [comment]
There's also something with GitHub actions runner being unable to fetch packages (totally unrelated to this PR).

File: ext/include/opentelemetry/ext/zpages/threadsafe_span_data.h (hunk 2 above)
Comment (pyohannes) [reply ext/include/opentelemetry/ext/zpages/threadsafe_span_data.h:179]: Yes, I think we can remove it if it's just one occurrence.

**Comment by reyang** [comment]
@xukaren please rebase to the latest master branch.

File: sdk/include/opentelemetry/sdk/trace/recordable.h (hunk 3 above)
Comment (xukaren) [reply sdk/include/opentelemetry/sdk/trace/recordable.h:86]: Should this be part of the scope resolution usage PR discussed [here](https://github.com/open-telemetry/opentelemetry-cpp/pull/363#discussion_r505600440)?

File: api/test/metrics/noop_metrics_test.cc (hunk 1 above)
Comment (xukaren) [reply api/test/metrics/noop_metrics_test.cc:44]: Thanks for the feedback! I've filed an issue for this here: #368

File: sdk/include/opentelemetry/sdk/trace/recordable.h (hunk 3 above)
Comment (xukaren) [reply sdk/include/opentelemetry/sdk/trace/recordable.h:87]: @ThomsonTan Could this be addressed as part of issue [#368 ](https://github.com/open-telemetry/opentelemetry-cpp/issues/368)?

**Comment by xukaren** [comment]
Rebased!

**Pull request merged**